src/cuga/backend/tools_env/registry/registry/api_registry.py

```
@@ -179,6 +227,35 @@ async def call_function(
+                        f"🛡️ Tool guard blocked call to '{function_name}': {error_message}"
+                    )
+                    return {
+                        "status": "exception",
+                        "status_code": 403,
+                        "message": f"Tool guard policy violation: {error_message}",
+                        "error_type": "ToolGuardViolation",
+                        "function_name": function_name,
+                    }
+                else:
+                    logger.debug(f"✅ Tool guard validation passed for '{function_name}'")
+            except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)